lib/fileio/translators/translatorMesh.cpp

```diff
@@ -55,191 +38,32 @@
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    status = dgMod.doIt();
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    // Add the Maya object to the point based deformer node's set.
-    const MFnGeometryFilter geomFilterFn(pointBasedDeformerNode, &status);
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    MObject deformerSet = geomFilterFn.deformerSet(&status);
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    MFnSet setFn(deformerSet, &status);
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    status = setFn.addMember(mayaObj);
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    // When we created the point based deformer, Maya will have automatically
-    // created a tweak deformer and put it *before* the point based deformer in
-    // the deformer chain. We don't want that, since any component edits made
-    // interactively in Maya will appear to have no effect since they'll be
-    // overridden by the point based deformer. Instead, we want the tweak to go
-    // *after* the point based deformer. To do this, we need to dig for the
-    // name of the tweak deformer node that Maya created to be able to pass it
-    // to the reorderDeformers command.
-    const MFnDagNode dagNodeFn(mayaObj, &status);
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    // XXX: This seems to be the "most sane" way of finding the tweak deformer
-    // node's name...
-    const std::string findTweakCmd = TfStringPrintf(
-        "from maya import cmds; [x for x in cmds.listHistory(\'%s\') if cmds.nodeType(x) == \'tweak\'][0]",
-        dagNodeFn.fullPathName().asChar());
-
-    MString tweakDeformerNodeName;
-    status = MGlobal::executePythonCommand(findTweakCmd.c_str(),
-                                           tweakDeformerNodeName);
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    // Do the reordering.
-    const std::string reorderDeformersCmd = TfStringPrintf(
-        "from maya import cmds; cmds.reorderDeformers(\'%s\', \'%s\', \'%s\')",
-        tweakDeformerNodeName.asChar(),
-        newPointBasedDeformerName.asChar(),
-        dagNodeFn.fullPathName().asChar());
-    status = MGlobal::executePythonCommand(reorderDeformersCmd.c_str());
-    CHECK_MSTATUS_AND_RETURN(status, false);
-
-    return true;
-}
-
-/* static */
-bool
-UsdMayaTranslatorMesh::Create(
-        const UsdGeomMesh& mesh,
-        MObject parentNode,
-        const UsdMayaPrimReaderArgs& args,
-        UsdMayaPrimReaderContext* context)
+MAYAUSD_NS_DEF {
+
+TranslatorMeshRead::TranslatorMeshRead(const UsdGeomMesh& mesh, 
+                                       const UsdPrim& prim, 
+                                       const MObject& transformObj,
+                                       const MObject& stageNode,
+                                       const GfInterval& frameRange,
+                                       bool wantCacheAnimation,
+                                       MStatus * status)
```

Comment:
It looks like `status` *could* be passed `nullptr` so the sites where we dereference it below should probably check it first?:
```
if (status != nullptr) {
```

Or we could store it into the local `stat` first, and then check before any early return.

Reply:
Awesome! Thanks @mattyjams

Good point, I initialized the status before passing it in to both meshRead and UsdMayaTranslatorUtil::CreateTransformNode. Please see 11b49c5c